api/notd/block_processor.py

```diff
@@ -71,32 +81,18 @@ async def process_block(self, blockNumber: int) -> List[RetrievedTokenTransfer]:
         # NOTE(krishan711): some blocks are too large to be retrieved from the AWS hosted node e.g. #14222802
         # for these, we can use infura specifically but if this problem gets too big find a better solution
         blockData = await self.ethClient.get_block(blockNumber=blockNumber, shouldHydrateTransactions=True)
-        retrievedTokenTransfers = []
-        erc721events = await self.ethClient.get_log_entries(startBlockNumber=blockNumber, endBlockNumber=blockNumber, topics=[self.erc721TansferEventSignatureHash])
-        logging.info(f'Found {len(erc721events)} erc721 events in block #{blockNumber}')
-        for event in erc721events:
-            retrievedTokenTransfers += await self._process_erc721_single_event(event=dict(event), blockData=blockData)
-        erc1155events = await self.ethClient.get_log_entries(startBlockNumber=blockNumber, endBlockNumber=blockNumber, topics=[self.erc1155TansferEventSignatureHash])
-        logging.info(f'Found {len(erc1155events)} erc1155Single events in block #{blockNumber}')
-        erc1155Transfers = []
-        for event in erc1155events:
-            erc1155Transfers += await self._process_erc1155_single_event(event=dict(event), blockData=blockData)
-        erc1155Batchevents = await self.ethClient.get_log_entries(startBlockNumber=blockNumber, endBlockNumber=blockNumber, topics=[self.erc1155TansferBatchEventSignatureHash])
-        logging.info(f'Found {len(erc1155Batchevents)} erc1155Batch events in block #{blockNumber}')
-        for event in erc1155Batchevents:
-            erc1155Transfers += await self._process_erc1155_batch_event(event=dict(event), blockData=blockData)
-        # NOTE(krishan711): these need to be merged because of floor seeps e.g. https://etherscan.io/tx/0x88affc90581254ca2ceb04cefac281c4e704d457999c6a7135072a92a7befc8b
-        retrievedTokenTransfers += await self._merge_erc1155_transfers(erc1155Transfers=erc1155Transfers)
-        blockNumber = blockData['number']
+        retrievedTokenTransfers =[]
+        retrievedTokenTransfers += await asyncio.gather(*[self.process_transaction(transaction=transaction) for transaction in blockData['transactions']])
```

Comment:
why do you delcare retrievedTokenTransfers as [] and then add to it? why not just set it directly to this result?